src/LM/counter.jl

```diff
@@ -0,0 +1,36 @@
+using DataStructures
+import DataStructures.Accumulator
+import DataStructures.DefaultDict
+import DataStructures.counter
+
+function normalize(accum)
+    #sum all counts
+    s = float(sum(accum))
+    #tuple of string with each count divided by sum
+    [(history,float(sum(count))/s) for (history,count) in accum]
+end
+
+function counter1(data, min::Integer, max::Integer,norm::Function;gamma = nothing)
+    data = (everygram(data,min_len = min, max_len =max ))
+    data = split.(data)
+    temp_lm = DefaultDict{SubString{String}, Accumulator{String,Int64}}(counter(SubString{String}))
```

Comment:
Please go through this: https://docs.julialang.org/en/v1/manual/style-guide/index.html#Avoid-writing-overly-specific-types-1
Try to minimize overly specified types if possible.

Reply:
ayush for using **Accumulator and counter** we have to overspecified at least her.